src/persist-client/src/lib.rs

```diff
@@ -675,6 +676,25 @@ impl PersistClient {
         &self,
         shard_id: &ShardId,
     ) -> Result<impl serde::Serialize, anyhow::Error> {
+        self.shard_state::<T>(shard_id)
+            .await
+            .and_then(|state| state.ok_or_else(|| anyhow::anyhow!("{} does not exist", shard_id)))
+    }
+
+    /// Returns the most recent applier version that was used to write to this shard.
+    pub async fn shard_version<T: Timestamp + Lattice + Codec64>(
```

Comment:
Hmm, instead of adding to the public API of persist, can we sniff the version out of the `inspect_shard` output?

Reply:
I can do that. The doc comment of `inspect_shard` scared me a little

```
    /// We'll be thoughtful about making unnecessary changes, but the **output
    /// of this method needs to be gated from users**, so that it's not subject
    /// to our backward compatibility guarantees.
```

but if you think it's Ok, then I'll do that.

Reply:
Yeah, I think it's _possible_ that the `inspect_shard` output would change, but as long as the catalog debug tools fail loudly when they can't find/understand the version, then I wouldn't be too stressed about it. There's definitely something appealing to me about reusing the existing inspect stuff for this over adding a new method